File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer,Integer> prev= new HashMap<>();
        for(int i=0;i<numbers.length;i++)
        {
            int num=numbers[i];
            int diff=target-num;
            
                if(prev.containsKey(diff))
                {
                    return new int[]{prev.get(diff)+1,i+1};
                }
                prev.put(num,i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            num=nums[i]
            diff=target-num
            if diff in map:
                return [map[diff],i]

            map[num]=i